### ISSUE TITLE
Centos 7 install fails, default version of git lacks the features for nvm install.sh

### ISSUE THREAD

**Issue by viper233** [fubarhouse/fubarhouse.npm#57]
``` TASK [nvm : NVM | Install] *********************************************************************************************************
fatal: [10.0.0.11]: FAILED! => {"changed": true, "cmd": "~/.nvm/install.sh", "delta": "0:00:00.981281", "end": "2018-05-07 16:53:59.559854", "failed": true, "rc": 1, "start": "2018-05-07 16:53:58.578573", "stderr": "Failed to update nvm, run 'git fetch' in /root/.nvm yourself.", "stderr_lines": ["Failed to update nvm, run 'git fetch' in /root/.nvm yourself."], "stdout": "=> nvm is already installed in /root/.nvm, trying to update using git\n\r=> ", "stdout_lines": ["=> nvm is already installed in /root/.nvm, trying to update using git", "", "=> "]}

PLAY RECAP **************************************************************************************************************************************************
10.0.0.1               : ok=62   changed=35   unreachable=0    failed=1
```
From the nvm install.sh file, the git fetch fails
```
git --git-dir=.git --work-tree=. fetch origin tag "v0.33.8" --depth=1
fatal: git fetch-pack: expected shallow list
```

This [issue](https://issues.jenkins-ci.org/browse/JENKINS-37229) has come up in other circles.

Not sure what the best route is around this for your role, I just wanted to let others know this is a problem.

Some suggestions
* Have a dependency on an [IUS](https://ius.io/) (e.g. linuxhq.ius) and _latest-git-centos7_ role.
* ignore it (probably best) and put a dependency on a later version of git as a requirement for the role.

**Comment by fubarhouse** [comment]
@viper233,

I wasn't notified of this issue, so I'll look into it shortly.

Sorry for the delay in starting this.

**Comment by fubarhouse** [comment]
Referencing https://github.com/creationix/nvm/issues/1772, It appears NVM has directly addressed this, I'd be interested in your feedback.

* [NVM Issue](https://github.com/creationix/nvm/issues/1772)
* [NVM PR](https://github.com/creationix/nvm/pull/1793)
* [Commit fix contained in PR above](https://github.com/creationix/nvm/pull/1793/commits/04c27e23fe82ff2edc7b3ff7f87fa191ad3e97e0)
* Looking at the release, it appears the documentation has been fixed in release [0.33.11](https://github.com/creationix/nvm/releases/tag/v0.33.11).

If the fix is simply to add a `git fetch`, I'll draft up that change tonight.

Would it be best to include this task exclusively on CentOS distributions only? - is there any negative impact on any of the other ~20 distributions currently supported that you can identify?

**Comment by fubarhouse** [comment]
@viper233, see the fetch task below - let me know if this helps and I'll commit it shortly. It's passing all the tests including idempotence - so there's basically no reason it can't be moved in.

`Affects tasks/nvm.yml line 122...`

```yaml
- name: "NVM | Clone"
  git:
    repo: "https://github.com/creationix/nvm.git"
    dest: "{{ nvm_install_dir }}"
    clone: yes
    update: no
    version: "{{ nvm_version }}"
  when: install_nvm_git|bool == true

- name: "NVM | Fetch"
  shell: "cd {{ nvm_install_dir }} && git fetch --tags origin"
  when: install_nvm_git|bool == true and fubarhouse_npm_nvm_installed.stat.exists|bool == false

- name: "NVM | Install"
  shell: "{{ nvm_install_dir }}/install.sh"
  when: fubarhouse_npm_nvm_installed.stat.exists|bool == false
```

**Comment by fubarhouse** [comment]
This is now merged at 8b826201b5d0aaecf339ef420672fb49c56d1311

If this needs to be reopened please feel free to do so.

#76 has prompted this one to be merged as I was waiting for feedback here.
